server/src/microservices/processors/metadata-extraction.processor.ts

```diff
@@ -293,7 +304,7 @@ export class MetadataExtractionProcessor {
         exposureTime: tags.ExposureTime ?? null,
         fileSizeInByte: stats.size,
         fNumber: validate(tags.FNumber),
-        focalLength: validate(tags.FocalLength),
+        focalLength: validate(tags.FocalLength as unknown as number | undefined), // Exiftool Tags type fails to incorporated `numericTags`.
```

Comment:
Maybe we can instead modify `ImmichTags` to handle this.

Reply:
I went down that path, but since `ImmichTags` is passed in to the `exifTool.read` generic, `ImmichTags` has to extend `Tags`. Changing `FocalLength` to `string` makes it an invalid extension. It would be nice if ExifTools were able to extend their own `Tags` type based on `numericTags`, but I can see how that would be extremely difficult to accomplish.

Reply:
Right, but those tags are generated to begin with and not even 100% accurate. I'd be more inclined to cast to unknown/any there and get the rights tags back for everything else.

Reply:
We can, for example extend `Omit<Tags, 'FocalLength'>` and then cast `read(...) as ImmichTags` or similar.

Reply:
You'd have to cast both calls to `read` for the variables `mediaTags` (:280-283), `sidecarTags` (:285-289), as well as casting `tags` back to Tags for the call to `firstDateTime` on line :301. 

But I'm not sure I understand your concern about the change above. The cast is passed into a function that will immediately confirm the cast is accurate. Further, this type is not exported, so the area of impact is limited to this single file.

Reply:
The main thought was to just have accurate typing. `Tags` is inaccurate when you calls `validate` and it would be nice if it was.

Reply:
I'll make that change. But I'm going to create an issue with exiftool-vendored, regarding `numericTags`